### ISSUE TITLE
Nested path matching

### ISSUE THREAD

**Issue by v79** [CarrotCodes/Pellet#76]
Hi,

I wonder if pellet could be extended to support path grouping, along the lines of:

```kotlin
path("/user") {
  get("/",::getUser)  // <- matches path /user/
  put("/new",::addUser) // <- matches path /user/new
}
```

Not sure if that sort of thing is a goal of Pellet or not, but I like to be able to group routes like this.

With thanks!

**Comment by CarrotCodes** [comment]
I do like the nested style offered by other frameworks, like Ktor, so this sounds like a good improvement.

I'm not sure if it'll be the first thing I work on, but it's on my list 👍

**Comment by CarrotCodes** [comment]
I was playing around with this, this morning, and came up with:

```kotlin
val idGroup = PelletHTTPRoutePath.Builder()
    .addComponents("/v1")
    .addVariable(idDescriptor)
    .build()
val sharedRouter = httpRouter {
    get("/", ::handleRequest)
    path("/v1") {
        get("/hello", ::handleResponseBody)
        post("/echo", ::handleEchoRequest)
        get("/error", ::handleForceError)
    }
    path(idGroup) {
        get("/hello", ::handleNamedResponseBody)
    }
}
```

Which results in the following route table:
```
GET / -> ::handleRequest
GET /v1/hello -> ::handleResponseBody
POST /v1/echo -> ::handleEchoRequest
GET /v1/error -> ::handleForceError
GET /v1/{id}/hello -> ::handleNamedResponseBody
```

I wanted to make sure that you could still do type-safe path groupings using `PelletHTTPRoutePath.Builder()`, as well as doing "raw paths".

Going to take some time to test it properly today and then I'll do a release.

**Comment by CarrotCodes** [comment]
Released in v0.0.14 🎉
